packages/design-system/src/components/VerticalNav/VerticalNavItem.tsx

```diff
@@ -88,7 +88,7 @@ export const VerticalNavItem = (props: VerticalNavItemProps): React.ReactElement
     setCollapsed(!collapsed);
 
     if (props.onSubnavToggle) {
-      props.onSubnavToggle(props.id, collapsed);
+      props.onSubnavToggle(id, collapsed);
```

Comment:
Updated because `props.id` is an optional prop. If id is not provided as a prop, this parameter value is `undefined`. By updating to use the function variable `id`, there will always be a value passed to the callback

Reply:
Good catch 👍